modules/e2ee/OlmAdapter.ts

```diff
@@ -876,46 +817,63 @@ export class OlmAdapter extends Listenable {
                         this._olmAccount.remove_one_time_keys(session_inbound);
                         olmData.session_for_reciving = session_inbound;
 
-                        const { key, index } = this._decryptKeyInfo(
-                            olmData.session_for_reciving,
-                            ciphertext,
+                        const decapsulatedSecret = await decapsulateSecret(
+                            encapsKyber,
+                            this._privateKyberKey,
                         );
 
-                        const pqKey = await decryptKeyInfoPQ(
-                            pqCiphertext,
-                            olmData.pqSessionKey,
+                        await olmData.setPQSessionKey(
+                            olmData.kemSecret,
+                            decapsulatedSecret,
                         );
 
-                        logger.info(`E2E: Recived new keys from ${pId}`);
-                        this._onKeysUpdated(pId, key, pqKey, index);
+                        const { key, index } =
+                            olmData.decryptKeyInfo(ciphertext);
 
-                        const olmCiphertext = this._encryptKeyInfo(
-                            olmData.session_for_sending,
-                        );
+                        const pqKey =
+                            await olmData.decryptPQKeyInfo(pqCiphertext);
 
-                        const pqCiphertextBase64 = await encryptKeyInfoPQ(
-                            olmData.pqSessionKey,
-                            this._mediaKeyPQ,
-                        );
-                        logger.info(`E2E: Sent my keys to ${pId}.`);
-                        this._sendSessionAckMessage(
-                            pqCiphertextBase64,
-                            olmCiphertext,
+                        const commitment = await commitToSecret(
                             pId,
+                            key,
+                            pqKey,
+                            index,
                         );
 
-                        olmData.status = PROTOCOL_STATUS.DONE;
-                        logger.info(
-                            `E2E: Participant ${pId} established E2E channel with us.`,
-                        );
+                        if (olmData.commitment != commitment)
+                            this._sendError(
+                                pId,
+                                `Keys do not match the commitment.`,
+                            );
+                        else {
```

Comment:
nit: maybe better to add the braces at if condition to mantain consistency and ease of read